ci/azure-pipelines.yml

```diff
@@ -4,6 +4,7 @@ variables:
   # Specify build author as an environment variable because on Apple Silicon
   # build_librepcb.sh is run locally where we want to use a different author.
   LIBREPCB_BUILD_AUTHOR: "LibrePCB CI"
+  CARGO_INCREMENTAL: 0  # Probably required for sccache(?)
```

Comment:
Since you usually have a clean build anyways in CI incremental compilation has no benefits.

Reply:
Hmm honestly I'm a bit confused about the incremental build. I read that it creates additional artifacts which are useless on CI *because* every build is a clean build. For example at [Swatinem/rust-cache](https://github.com/Swatinem/rust-cache) they write:

> In particular, the workspace crates themselves are not cached since doing so is generally not effective. For this reason, this action automatically sets CARGO_INCREMENTAL=0 to disable incremental compilation, so that the Rust compiler doesn't waste time creating the additional artifacts required for incremental builds.

So far I'm not sure what is the best way to speed up CI. My local tests with `sccache` were not successful at all, it did not speed up rebuilds. I even experienced CMake to hang randomly which *might* be related to `sccache`. [Swatinem/rust-cache](https://github.com/Swatinem/rust-cache) sounds interesting to cache dependencies, but they write it's not speeding up building the own crate.

The Fluvio project wrote a [blog post about using sccache on CI](https://www.infinyon.com/blog/2021/04/github-actions-best-practices/) but afterwards they [removed sccache from CI](https://github.com/infinyon/fluvio/commit/7c55089253d5819f3e53de1634c6c50ac42e67cd).

After all, it sounds like caching Rust build artifacts is rather difficult. Maybe we should go with [Swatinem/rust-cache](https://github.com/Swatinem/rust-cache) for now (no sccache) as so far dependencies are a much bigger part than our own Rust code.

Reply:
I never really looked into sccache. But for rust incremental compilation: The smallest compilation unit in Rust is a crate (as opposed to C where it is a file). But with incremental compilation the compiler caches with very fine granularity which speeds up small changes a lot. But it need to generate and store all the required metadata and intermediate artifacts, which makes building from a clean state slightly slower. More information on incremental compilation:
 * https://blog.rust-lang.org/2016/09/08/incremental.html
 * https://lwn.net/Articles/997784/
 * https://perf.rust-lang.org/dashboard.html
 * https://rustc-dev-guide.rust-lang.org/queries/incremental-compilation-in-detail.html

Reply:
In https://github.com/rnestler/reboot-arch-btw/blob/master/.github/workflows/ci.yml I just cache the `target` folder and the registry cache.

Reply:
Thanks for the explanation, that halps. And the performance dashboard is nice :)

I just wonder, did you read the code change exactly? It sounds like you argue *against* using incremental builds while the change actually disables incremental builds.

Reply:
> It sounds like you argue against using incremental builds while the change actually disables incremental builds.

Yes. My initial comment should have an "anyway" in the end. So not only because of sscache you'd want to disable it but also because it's usually not worth to do in CI.

Reply:
:+1: Improved the comment accordingly.